ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1486,6 +1519,14 @@ void checkAndMaybeComplete() {
       }
     }
 
+    /**
+     * @return {@code true} If this chunk is a metadata chunk and router reserved metadata is enabled. {@code false}
+     * otherwise.
+     */
+    boolean shouldUseReservedMetadataId() {
+      return this instanceof MetadataPutChunk && routerConfig.routerReservedMetadataEnabled;
```

Comment:
nit: isMetadataChunk()

Reply:
Again, shall we use isMetadataChunk instead of "this instance of MetadataPutChunk"?